harness/go/harness/util.go

```diff
@@ -13,3 +15,17 @@ var RetryDisabled = &temporal.RetryPolicy{MaximumAttempts: 1}
 func AppErrorf(msg string, args ...interface{}) error {
 	return temporal.NewApplicationError(fmt.Sprintf(msg, args...), "SdkFeaturesError")
 }
+
+// Find the first event in the history that meets the condition.
+func FindEvent(history client.HistoryEventIterator, cond func(*historypb.HistoryEvent) bool) (*historypb.HistoryEvent, error) {
```

Comment:
Nice, now that we have this, can you rewrite the binary test to use it?

Reply:
yep ill add a commit it do that.